### DISCUSSION TITLE
What are x and y?

### DISCUSSION THREAD

**Discussion by ParanoidUser** [ParanoidUser/yolo#508] [Galaxy Trivia] [answered]
Two integers x and y are selected such that 1 < x <= y < 100.

Player A is given the product x*y and player B is given the sum x+y.

The following dialogue takes place:

A: I don’t know the values x and y.
B: I knew you didn’t know. I don’t know them either.
A: Now I know x and y.
B: Now I know them too.

**Comment by peterbaccaro** [comment]
x and y are letters of the alphabet. :grin:

**Comment by peterbaccaro** [comment]
x is 4 and y is 13.

**Reply by ParanoidUser** [reply]
Can you explain how these numbers were chosen? 🧮

**Reply by peterbaccaro** [reply]
1) **A: I don't know the values of x and y** means the product can be factored in more than one way so x and y cannot both be primes.

2) **B: I knew you didn’t know. I don’t know them either.** means that for every possible sum decomposition of the number x + y which A knows, the product of x * y has more than one product decomposition which prevents B from knowing x and y.

The conditions: 1 < x <= y < 100 means that the possible sums are from 5 to 100.
The prime numbers less than 100 are 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97.

Sums that can be formed by adding two primes can be eliminated so we are left with 24 possible sums: 11, 17, 23, 27, 29, 35, 37, 41, 47, 51, 57, 59, 65, 67, 71, 77, 79, 83, 87, 89, 93, 95, 97.

3) **A: Now I know x and y.** This means the number x * y has only one product decomposition.  Since each of the 24 sums is odd, we know that one of either x or y is odd and the other is even.

4) **B: Now I know them too.** This means the numbers x + y has only one sum decomposition.  16 of the 24 sums can be written in at least two different ways as a sum of a power of 2 and an odd prime therefore these can be eliminated.

This leaves 8 possible sums: 17, 29, 41, 53, 59, 65, 89, 97.
7 of these can be written in two different ways as the sum of a power of 2 and the product of any number of odd primes, therefore these can be eliminated.
This leaves one possible sum, 17.

When you look at the possible sum and product decompositions for 17 each product except 52 has at least two distinct factorizations into an odd an even factor, leading to an ambiguity
The product given to B must then be 52 which factorizes into a pair of even and odd factors as 
52 = 4 x 13
so...
**x = 4 and y = 13**

**Reply by ParanoidUser** [reply]
Bravo! 👏👏👏

**Comment by thuongtruong1009** [comment, marked as answer]
The integers x and y are 8 and 13.

Here's how to arrive at the solution:

First, let's consider all the possible products of two numbers between 1 and 99, excluding the products where x and y are equal (since 1 < x <= y):

2*2=4, 2*3=6, 2*4=8, ..., 2*98=196
3*4=12, 3*5=15, 3*6=18, ..., 3*98=294
4*5=20, 4*6=24, 4*7=28, ..., 4*98=392
...
97*98=9506

There are a total of 4851 products, and each of them can be expressed as a sum of two integers between 3 and 197 (the minimum and maximum values of x+y).

Now let's look at Player A's first statement: "I don’t know the values x and y." This means that there are at least two possible products that Player A could have been given. Specifically, there must be at least two products that have the same sum.

Let's list all the possible sums and see which products have the same sum:

3: None
4: None
5: None
6: None
7: None
8: 2*4, 3*5
9: None
10: 2*5, 4*6
11: 2*6, 5*6, 3*8
12: 2*7, 3*9, 4*8
13: 2*8, 3*10, 5*6
14: 2*9, 3*11, 4*10, 5*9
15: 2*10, 3*12, 4*11, 5*10, 6*9
16: 2*11, 3*13, 4*12, 5*11, 6*10
17: 2*12, 3*14, 4*13, 5*12, 6*11, 7*10
18: 2*13, 3*15, 4*14, 5*13, 6*12, 7*11, 8*10
19: 2*14, 3*16, 4*15, 5*14, 6*13, 7*12, 8*11
...
196: 97*99
197: 98*99

We can see that there are two possible products for the sum 21: 417 and 615. Therefore, Player A could have been given either of these products, and Player B could have been given the sum 21.

Now let's move on to Player B's first statement: "I knew you didn’t know. I don’t know them either." This means that there must be more than one possible sum that could have been given to Player B.

The only sums with multiple possible products are 10 and 21. If Player B had been given the sum 10, they would have known that the product could only be 3*7, since all the other products with a sum of 10 have unique factorizations. However, since Player B doesn't know the answer, we can conclude that Player A was given one of the products with a sum of 21.
